File: C++/0026.cpp
#include <iostream>
using namespace std;


int main()
{
    int x1, y1, r1, x2, y2, r2, squaredDistance;

    cin >> x1 >> y1 >> r1;
    cin >> x2 >> y2 >> r2;

    squaredDistance = (x1 - x2) * (x1 - x2) + (y1 - y2) * (y1 - y2);

    if (squaredDistance <= (r1 + r2) * (r1 + r2)
            && squaredDistance >= (r2 - r1) * (r2 - r1))
        cout << "YES" << endl;
    else
        cout << "NO" << endl;

    return 0;
}

File: Python/0026.py
x1, y1, r1 = map(int, input().split())
x2, y2, r2 = map(int, input().split())

squaredDistance = (x1 - x2) ** 2 + (y1 - y2) ** 2

if squaredDistance <= (r1 + r2) ** 2 and squaredDistance >= (r2 - r1) ** 2:
    print('YES')
else:
    print('NO')
